.github/workflows/cross-compile.yml

```diff
@@ -0,0 +1,115 @@
+            os: ubuntu-20.04
+            profile: 'x86_64'
+            packages: 'cmake g++'
+
+          - target: linux_arm
+            os: ubuntu-20.04
+            profile: 'armhf'
+            packages: 'cmake g++-arm-linux-gnueabihf'
+
+          - target: linux_arm64
+            os: ubuntu-20.04
+            profile: 'aarch64'
+            packages: 'cmake g++-aarch64-linux-gnu'
+
+          - target: linux_ppc
+            os: ubuntu-20.04
+            profile: 'ppc64'
+            packages: 'cmake g++-powerpc64le-linux-gnu'
+
+          - target: windows_32
+            os: ubuntu-20.04
+            profile: 'mingw32'
+            packages: 'cmake g++-mingw-w64-i686'
+
+          - target: windows_64
+            os: ubuntu-20.04
+            profile: 'mingw64'
+            packages: 'cmake g++-mingw-w64-x86-64'
+
+          - target: windows_arm64
+            os: ubuntu-20.04
+            profile: 'mingwaarch64'
+            packages: 'cmake clang'
+
+          - target: osx_64
+            os: macos-10.15
+            profile: 'x86_64'
+
+          - target: osx_arm64
+            os: macos-10.15
+            profile: 'aarch64'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup APT packages
+        if: ${{ matrix.os == 'ubuntu-20.04' }}
+        run: sudo apt-get install --yes ${{ matrix.packages }}
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: adopt
+      - name: Build with Maven
+        run: mvn -B clean install -P ${{ matrix.profile }}
+      - run: git config --global user.email "<>"
+      - run: git config --global user.name "Automated build on Github"
+      - run: git fetch && git checkout -t origin/${{ needs.create-branch.outputs.precompiled_branch }}
+      - run: git add src/main/resources-precompiled/**
+      - run: git commit --allow-empty -m "Precompiled natives (@${{ needs.create-branch.outputs.base_rev }}) for ${{ matrix.target }}"
+      - name: Push recompiled binary for ${{ matrix.profile }}
+        run: 'while  git pull --rebase && ! git push; do sleep 5; done'
+
+  compile-natives-solaris:
+    runs-on: macos-latest
+    needs: create-branch
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build on solaris
+        uses: vmactions/solaris-vm@v0.0.3
+        with:
+          envs: MAVEN_OPTS
+          run: |
+            wget -qO- https://download.bell-sw.com/java/11.0.12+7/bellsoft-jdk11.0.12+7-solaris-x64-lite.tar.gz | tar xvf -
+            wget -qO- https://downloads.apache.org/maven/maven-3/3.8.1/binaries/apache-maven-3.8.1-bin.tar.gz | gunzip - | tar xvf -
+            pkg install gcc git cmake
+            JAVA_HOME=./jdk-11.0.12-lite ./apache-maven-3.8.1/bin/mvn -B clean install -P x86 -DskipTests
+            JAVA_HOME=./jdk-11.0.12-lite ./apache-maven-3.8.1/bin/mvn -B clean install -P x86_64 -DskipTests
```

Comment:
VM setup is to heavy to do it twice, so we build both 32 and 64 in the same workspace with only `mvn clean` in-between.